src/core/load-dropin.c

```diff
@@ -25,10 +25,14 @@ static int process_deps(Unit *u, UnitDependency dependency, const char *dir_suff
                 return r;
 
         STRV_FOREACH(p, paths) {
-                _cleanup_free_ char *target = NULL;
-                const char *entry;
+                _cleanup_free_ char *target = NULL, *target_file = NULL, *entry = NULL;
 
-                entry = basename(*p);
+                r = path_extract_filename(*p, &entry);
```

Comment:
I think this should be moved to after `null_or_empty_path()` and `is_symlink()` checks below.